src/base/rss/rss_autodownloadrule.cpp

```diff
@@ -117,13 +122,53 @@ namespace RSS
                     && (lastMatch == other.lastMatch)
                     && (savePath == other.savePath)
                     && (category == other.category)
-                    && (addPaused == other.addPaused);
+                    && (addPaused == other.addPaused)
+                    && (smartFilter == other.smartFilter)
+                    && (previouslyMatchedEpisodes == other.previouslyMatchedEpisodes);
         }
     };
 }
 
 using namespace RSS;
 
+static const QString episodeRegex = "(?:^|[^A-Za-z0-9])((?:S\\d{1,}(?:E\\d{1,})+)|(?:\\d{1,}x\\d{1,})|(?:\\d{4}\\.\\d{2}\\.\\d{2}))(?:[^A-Za-z0-9]|$)";
+
+QString computeEpisodeName(const QString &article)
+{
+    QRegExp f(episodeRegex, Qt::CaseInsensitive);
```

Comment:
> Also, make episodeRegex an actual QRegularExpression object, don't initialize it here every time.

Disagree: those who do not use this feature shall not pay for it with their memory.